barretenberg/cpp/src/barretenberg/relations/utils.hpp

```diff
@@ -147,23 +149,21 @@ template <typename Flavor> class RelationUtils {
      * relation. This value is checked against the final value of the target total sum (called sigma_0 in the
      * thesis).
      */
-    template <typename Parameters, size_t relation_idx = 0>
+    template <typename Parameters>
     // TODO(#224)(Cody): Input should be an array?
-    inline static void accumulate_relation_evaluations_without_skipping(PolynomialEvaluations evaluations,
+    inline static void accumulate_relation_evaluations_without_skipping(const PolynomialEvaluations& evaluations,
                                                                         RelationEvaluations& relation_evaluations,
                                                                         const Parameters& relation_parameters,
                                                                         const FF& partial_evaluation_result)
     {
-        using Relation = std::tuple_element_t<relation_idx, Relations>;
-
-        Relation::accumulate(
-            std::get<relation_idx>(relation_evaluations), evaluations, relation_parameters, partial_evaluation_result);
-
-        // Repeat for the next relation.
-        if constexpr (relation_idx + 1 < NUM_RELATIONS) {
-            accumulate_relation_evaluations<Parameters, relation_idx + 1>(
-                evaluations, relation_evaluations, relation_parameters, partial_evaluation_result);
-        }
+        auto seq = std::make_index_sequence<NUM_RELATIONS>{};
+        // This evaluates the function across the relations, without recursion.
+        [&]<size_t... I>(std::index_sequence<I...>) {
+            // FIXME: You wan't /*consider_skipping=*/false here, but tests need to be fixed.
```

Comment:
Not sure I get this, and why not fix the test if so?

Reply:
I don't have the time, like 20 verification tests fail. Cody said Luke will look at it.

Reply:
Yeah it wasn't immediately obvious to me what the issue is so I thought Luke should have a look.